libs/components/forms/src/lib/modules/character-counter/character-counter-indicator.component.ts

```diff
@@ -11,29 +11,30 @@ import {
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class SkyCharacterCounterIndicatorComponent {
-  public characterCountMessage: string;
-  public overLimitMessage: string;
+  #_currentCharacterCountLimit = 0;
+  #_currentCharacterCount = 0;
 
-  private currentCharacterCountLimit = 0;
-  private currentCharacterCount = 0;
+  #changeDetector: ChangeDetectorRef;
 
-  constructor(private changeDetector: ChangeDetectorRef) {}
+  constructor(changeDetector: ChangeDetectorRef) {
+    this.#changeDetector = changeDetector;
+  }
 
   public get characterCount(): number {
-    return this.currentCharacterCount;
+    return this.#_currentCharacterCount;
   }
 
   public set characterCount(count: number) {
-    this.currentCharacterCount = count;
-    this.changeDetector.markForCheck();
+    this.#_currentCharacterCount = count;
+    this.#changeDetector.markForCheck();
   }
 
   public get characterCountLimit(): number {
-    return this.currentCharacterCountLimit;
+    return this.#_currentCharacterCountLimit;
   }
 
   public set characterCountLimit(limit: number) {
-    this.currentCharacterCountLimit = limit;
-    this.changeDetector.markForCheck();
+    this.#_currentCharacterCountLimit = limit;
```

Comment:
☝️ Since we're renaming these properties, we might as well match them to their getter/setters. `#_characterCountLimit`.

Reply:
Also, come to think of it, why not make these properties bonafide `@Input()` (with `@internal` applied) so that we don't need to use getter/setters with change detection?

Reply:
I have updated the names; however, I can't update to a true `Input` because this component isn't created using a template and is updated using typescript so the `markForCheck` calls are needed.

Reply:
That's odd. No problem.